src/coreComponents/mesh/FaceManager.cpp

```diff
@@ -69,497 +70,76 @@ void FaceManager::resize( localIndex const newSize )
-  GEOSX_ERROR_IF_NE( numUniqueFaces, elemSubRegionList.size( 0 ) );
-  GEOSX_ERROR_IF_NE( numUniqueFaces, elemList.size( 0 ) );
-  GEOSX_ERROR_IF_NE( numUniqueFaces, nodeList.size() );
-
-  // loop over all the nodes.
-  forAll< parallelHostPolicy >( numNodes, [&]( localIndex const nodeID )
-  {
-    localIndex curFaceID = uniqueFaceOffsets[ nodeID ];
-    localIndex const numFaces = facesByLowestNode.sizeOfArray( nodeID );
-
-    // loop over all the FaceBuilders associated with the node
-    localIndex j = 0;
-    for(; j < numFaces - 1; ++j )
-    {
-      // If two subsequent FaceBuilders compare equal then they describe an interior face.
-      if( facesByLowestNode( nodeID, j ) == facesByLowestNode( nodeID, j + 1 ) )
-      {
-        addInteriorFace( elementManager, curFaceID, facesByLowestNode( nodeID, j ), facesByLowestNode( nodeID, j + 1 ), elemRegionList, elemSubRegionList,
-                         elemList, nodeList );
-        ++j;
-      }
-      // Otherwise it's a boundary face.
-      else
-      {
-        addBoundaryFace( elementManager, curFaceID, facesByLowestNode( nodeID, j ), elemRegionList, elemSubRegionList, elemList, nodeList );
-      }
-
-      ++curFaceID;
-    }
-
-    if( j == numFaces - 1 )
-    {
-      addBoundaryFace( elementManager, curFaceID, facesByLowestNode( nodeID, j ), elemRegionList, elemSubRegionList, elemList, nodeList );
-    }
-  } );
-}
-
-
-void FaceManager::buildFaces( NodeManager & nodeManager, ElementRegionManager & elementManager )
+void FaceManager::buildFaces( CellBlockManagerABC const & cellBlockManager,
+                              NodeManager & nodeManager,
+                              ElementRegionManager & elementManager )
 {
   GEOSX_MARK_FUNCTION;
 
   m_toElements.setElementRegionManager( elementManager );
 
-  localIndex const numNodes = nodeManager.size();
+  resize( cellBlockManager.numFaces() );
 
-  ArrayOfArrays< FaceBuilder > facesByLowestNode( numNodes, 2 * maxFacesPerNode() );
-  createFacesByLowestNode( elementManager, facesByLowestNode.toView() );
+  populateRegions( elementManager,
+                   m_toElements.m_toElementRegion,
+                   m_toElements.m_toElementSubRegion );
 
-  array1d< localIndex > uniqueFaceOffsets( numNodes + 1 );
-  localIndex const numFaces = calculateTotalNumberOfFaces( facesByLowestNode.toViewConst(), uniqueFaceOffsets );
-
-  resizeFaceToNodeMap( elementManager,
-                       facesByLowestNode.toViewConst(),
-                       uniqueFaceOffsets,
-                       nodeList() );
-
-  resize( numFaces );
-
-  populateMaps( elementManager,
-                facesByLowestNode.toViewConst(),
-                uniqueFaceOffsets,
-                m_toElements.m_toElementRegion,
-                m_toElements.m_toElementSubRegion,
-                m_toElements.m_toElementIndex,
-                nodeList() );
+  m_toElements.m_toElementIndex = cellBlockManager.getFaceToElements(); // TODO do better
```

Comment:
Why do better ? Do you think of a better alternative ?

Reply:
Because for the moment I'm just copying data with `operator=` and maybe something better is possible with move semantics.

Also, this is surely possible to define an overallocation policy, and this is potentially the place to do this.
We can do it directly in the `CellBlockManager*` region too. To be discussed.